### PR TITLE
feat(@desktop/wallet): Implementation of Sign Swap

### LINKED ISSUE #14835: Sign Swap
https://www.notion.so/Sign-Swap-20a4991793f0409085f6a4bd2cba9e84
https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3542-498146&t=eTvhBjNa4YYvdRU7-0

- Get data from Swap proposal
- Trigger Swap Tx
- Auto refresh bottom bar data

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #14835

### What does the PR do

Implements the Swap and Sign flow for Swaps -> https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3511-433394&t=dlm3FaxR05MRt7xX-4
UserStory -> https://www.notion.so/Sign-Swap-20a4991793f0409085f6a4bd2cba9e84?pvs=4

Note this is not actual UI. Actual UI is being developed under https://github.com/status-im/status-desktop/issues/14785

### Affected areas

SwapModal.qml, SwapSignApprovePopup.qml

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Storybook:
Screenshot 2024-07-02 at 12 33 45 AM

Swap Flow:

https://github.com/status-im/status-desktop/assets/60327365/b56a3a6a-2f95-4679-a4c6-14074b0daaa3

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml
Comment (dlipicar) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:336]: Hmm now I'm wondering if we need to show the total fees here (for approval+swap) or just the approval ones until we're able to perform the swap...

**Comment by dlipicar** [comment]
image
hmm this looks silly. There's nothing about handling long numbers in the designs, is there? We might want to ask Alisher.

**Review by dlipicar — APPROVED** [review]
changes look good!

**Comment by alexjba** [comment]
I think the swap modal should close once we've signed the transaction 🤔

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:336]: hmm thats a  good point, I'll ask Alisher

**Comment by Khushboo-dev-cpp** [comment]
> at the 0:02 mark of the video, the input number shouldn't be red when the user has just opened the Swap Modal, right? I've noticed this while testing as well.

good point! I'll raise a bug for it as I am not sure what causes this and we may need to update tests etc. 
Ive added it to the bug I raised https://github.com/status-im/status-desktop/issues/15417

**Comment by Khushboo-dev-cpp** [comment]
> image hmm this looks silly. There's nothing about handling long numbers in the designs, is there? We might want to ask Alisher for the real UI component.

Ive asked here https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK?node-id=3542-498146#861439212

**Comment by Khushboo-dev-cpp** [comment]
> I think the swap modal should close once we've signed the transaction 🤔


https://github.com/status-im/status-desktop/assets/60327365/86d1e7e2-829b-4c6c-9659-ce884d640a8a

@alexjba thanks for pointing it out, its already done in the PR, I think I recorded the video Ive attached before I made the fix and hence is not visible but here's a recording of the correct behaviour :)

**Review by alexjba — APPROVED** [review]
LGTM!

File: ui/app/AppLayouts/Wallet/popups/swap/SwapOutputData.qml
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapOutputData.qml:12]: hmm good point, I actually don't know what to create for it.  perhaps will only happen after M1 release and the rework done by the core team on the send flow. cause its common for send and swap right now.

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:336]: https://github.com/status-im/status-desktop/issues/15439 created an issue to track this

**Pull request merged**
